src/modules/common/components/Headroom.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import raf from 'raf'
+
+import style from './Headroom.css'
+
+class Headroom extends React.PureComponent {
+  static propTypes = {
+    children: PropTypes.any.isRequired,
+    scrollHeight: PropTypes.number.isRequired,
+    pinStart: PropTypes.number.isRequired,
+    stickyAncestor: PropTypes.node,
+    height: PropTypes.number
+  }
+
+  static defaultProps = {
+    pinStart: 0
+  }
+
+  constructor (props) {
+    super(props)
+    this.handleEvent = this.handleEvent.bind(this)
+    this.update = this.update.bind(this)
+    this.ticking = false
+    this.lastScrollY = 0
+    this.lastTransform = 0
+    this.lastKnownScrollY = 0
+    this.currentScrollY = 0
+    this.direction = 'down'
```

Comment:
Magic number, introduce some kind of constant for this

Reply:
0 is not a magic number imo

Reply:
k, let's call it magic string 'down' :D